nas.py

```diff
@@ -69,6 +68,21 @@ def copy(self, remote_user, remote_ip, remote_path, local_path, to_local):
         stdout, stderr = p.communicate()
         return stdout, stderr, p.returncode
 
+    def copy_to_restricted(self, user, ip, from_path, to_path,
```

Comment:
Scp can't copy to user restriced locations directly. The program created sudo /tmp/net-applet-shuffler is not user owned. Thus, either the owner has to be changed, the function in question has to be used.